packages/matplotlib/src/html5_canvas_backend.py

```diff
@@ -0,0 +1,471 @@
+import numpy as np
+from matplotlib.backends.browser_backend import FigureCanvasWasm, NavigationToolbar2Wasm
+from matplotlib.backend_bases import (
+    GraphicsContextBase,
+    RendererBase,
+    FigureManagerBase,
+    _Backend,
+)
+
+from PIL import Image
+from PIL.PngImagePlugin import PngInfo
+
+from matplotlib import __version__
+from matplotlib.colors import colorConverter, rgb2hex
+from matplotlib.transforms import Affine2D
+from matplotlib.path import Path
+from matplotlib import interactive
+
+from matplotlib.cbook import maxdict
+from matplotlib.font_manager import findfont
+from matplotlib.ft2font import FT2Font, LOAD_NO_HINTING
+from matplotlib.mathtext import MathTextParser
+
+from pyodide import create_proxy
+
+from js import document, window, XMLHttpRequest, ImageData, FontFace
+
+import base64
+import io
+import math
+
+_capstyle_d = {"projecting": "square", "butt": "butt", "round": "round"}
+
+# The URLs of fonts that have already been loaded into the browser
+_font_set = set()
+
+if hasattr(window, "testing"):
+    _base_fonts_url = "/fonts/"
+else:
+    _base_fonts_url = "/pyodide/fonts/"
```

Comment:
This is from the earlier days of Iodide. Basically, if you download all the files related to Pyodide i.e. the `.data` and `.js` files and place them in a directory. Then, if you serve the directory, the path of `matplotlib.js` file would be `https://localhost:8000/matplotlib.js` and for fonts it would be `https://localhost:8000/fonts/cmb10.ttf`. This is what is used in tests as well, that's we the path is only `/fonts/` if we are using the tests.

However, back in the case of Iodide, we used to have all the `.data` and `.js` files in a directory known as `pyodide`. Suppose this `pyodide` directory is inside another directory (call it `A`) and now if we serve directory `A`, the paths for the files would become `https://localhost:8000/pyodide/matplotlib.js` and `https://localhost:8000/pyodide/fonts/cmb10.ttf`.

So this is assuming that when one deploys `pyodide`, instead of serving at the level where all `.data` and `.js` files exist, we are instead serving from one level up where the folder containing all the `.data` and `.js` files are inside a directory called `pyodide`.

But yes, we can change it. Since Iodide is deprecated now. Please let me know what should we choose here.